packages/contracts/contracts/voting/erc20/Erc20VotingManager.sol

```diff
@@ -0,0 +1,215 @@
+            GovernanceERC20(_tokenSetting.addr)
+        );
+
+        plugin = createProxy(dao, getImplementationAddress(), init);
+    }
+
+    /// @inheritdoc PluginManager
+    function getInstallPermissions(bytes memory data)
+        external
+        view
+        virtual
+        override
+        returns (RequestedPermission[] memory permissions, string[] memory helperNames)
+    {
+        (, TokenSetting memory _tokenSetting) = abi.decode(data, (VoteSetting, TokenSetting));
+
+        permissions = new RequestedPermission[](_tokenSetting.addr == address(0) ? 5 : 4);
+        helperNames = new string[](_tokenSetting.addr == address(0) ? 1 : 0);
+
+        address NO_ORACLE = address(0);
+
+        // Allows plugin to call DAO with EXEC_PERMISSION
+        permissions[0] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            DAO_PLACEHOLDER,
+            PLUGIN_PLACEHOLDER,
+            NO_ORACLE,
+            keccak256("EXECUTE_PERMISSION")
+        );
+
+        // Allows DAO to call plugin with SET_CONFIGURATION_PERMISSION
+        permissions[1] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.SET_CONFIGURATION_PERMISSION_ID()
+        );
+
+        // Allows DAO to call plugin with UPGRADE_PERMISSION
+        permissions[2] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.UPGRADE_PERMISSION_ID()
+        );
+
+        // Allows DAO to call plugin with SET_TRUSTED_FORWARDER_PERMISSION
+        permissions[3] = buildPermission(
+            BulkPermissionsLib.Operation.Grant,
+            PLUGIN_PLACEHOLDER,
+            DAO_PLACEHOLDER,
+            NO_ORACLE,
+            erc20VotingBase.SET_TRUSTED_FORWARDER_PERMISSION_ID()
+        );
+
+        uint256 GOVERNANCE_ERC20_HELPER_IDX = 0;
+
+        if (_tokenSetting.addr == address(0)) {
+            // Allows DAO to call relatedContracts (token) with MINT_PERMISSION.
+            permissions[4] = buildPermission(
+                BulkPermissionsLib.Operation.Grant,
+                GOVERNANCE_ERC20_HELPER_IDX,
+                DAO_PLACEHOLDER,
+                NO_ORACLE,
+                keccak256("MINT_PERMISSION")
+            );
+
+            helperNames[GOVERNANCE_ERC20_HELPER_IDX] = "GovernanceERC20";
+        }
+    }
+
+    /// @inheritdoc PluginManager
+    function getImplementationAddress() public view virtual override returns (address) {
+        return address(erc20VotingBase);
+    }
+
+    /// @inheritdoc PluginManager
+    function deployABI() external view virtual override returns (string memory) {
+        return
+            "((tuple(uint64,uint64,uint64) voteSetting),(tuple(address addr,string name,string symbol, (tuple(address[],uint256[]) mintSetting)) tokenSetting))";
```

Comment:
```suggestion
        return
            "(\
                (tuple(uint64,uint64,uint64) voteSetting),\
                (tuple(address addr,string name,string symbol,\
                (tuple(address[],uint256[]) mintSetting)) tokenSetting)\
            )";
```

Are line continuation characters ok to use?

Reply:
not sure, we can test that with testing